WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -199,6 +208,22 @@ final class EditableOrderViewModel: ObservableObject {
     ///
     @Published var selectedProductViewModel: ProductInOrderViewModel? = nil
 
+    /// Keeps track of selected/unselected Products, if any
+    ///
+    @Published var selectedProducts: [Product?] = []
+
+    /// Keeps track of selected/unselected Product Variations, if any
+    ///
+    @Published var selectedProductVariations: [ProductVariation?] = []
+
+    /// Keeps track of all selected Products and Product Variations IDs
+    ///
+    var selectedProductsAndVariationsIDs: [Int64] {
+        let selectedProductsCount = selectedProducts.compactMap { $0?.productID }
+        let selectedProductVariationsCount = selectedProductVariations.compactMap { $0?.productVariationID }
+        return selectedProductsCount + selectedProductVariationsCount
```

Comment:
As discussed in pairing – ideally we shouldn't run the lists of IDs together. This works because WordPress underlies it all and these are all Posts... but we shouldn't really rely on that because it's not a good understanding of the data. If needed, we can keep it even after the refactor, because it works and this is philosophical... but it does stand out as a strange way to handle the data.

Reply:
I've opened a new issue here to explore this in the future: https://github.com/woocommerce/woocommerce-ios/issues/9228 👍